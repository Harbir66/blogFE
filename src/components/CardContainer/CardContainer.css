main {
  flex: 1;
  /* display: flex;
  flex-flow: row wrap; */
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
  justify-content: space-evenly;
  margin:20px 0;
  gap:30px 80px;
}