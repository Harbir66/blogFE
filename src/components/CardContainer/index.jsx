/* eslint-disable camelcase */
import React from 'react';
import './CardContainer.css';
import { useNavigate } from 'react-router-dom';
import Card from '../Card';
import { GET_BLOG_DATA } from '../../constants/apiEndPoints';
import makeRequest from '../../utils/makeRequest';

function CardContainer() {
  const [cardData, setCardData] = React.useState();
  const [error, setError] = React.useState(null);
  const navigate = useNavigate();

  React.useEffect(() => {
    makeRequest(GET_BLOG_DATA, {}, navigate)
      .then((data) => {
        setCardData(data);
      })
      .catch((err) => {
        setError(err.message);
      });
  }, []);

  if (error) {
    return (
      <div>
        <p>{error}</p>
      </div>
    );
  }

  const cards = cardData ? (
    cardData.map((cardInfo) => {
      const {
        id,
        date,
        reading_time,
        title,
        description,
        claps,
        liked,
        image,
      } = cardInfo;
      return (
        <Card
          key={id}
          id={id}
          date={date}
          readingTime={reading_time}
          title={title}
          description={description}
          claps={claps}
          liked={liked}
          image={image}
        />
      );
    })
  ) : (
    <div data-testid="loading">
      <p>Loading...</p>
    </div>
  );
  return <main className="padding">{cards}</main>;
}

export default CardContainer;
