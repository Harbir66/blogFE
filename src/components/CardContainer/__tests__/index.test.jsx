import React from 'react';
import { render, waitFor, screen } from '@testing-library/react';
import CardContainer from '..';
import makeRequest from '../../../utils/makeRequest';

const mockData = [
  {
    id: 1,
    date: '2020-10-10T00:00:00.000Z',
    reading_time: '5 mins',
    title: 'Mock Title 1',
    description: 'This is a description',
    claps: 10,
    liked: false,
    image: 'https://i.ibb.co/LNxt44v/abstract.png',
  },
  {
    id: 2,
    date: '2020-10-10T00:00:00.000Z',
    reading_time: '5 mins',
    title: 'Mock Title 2',
    description: 'This is a description',
    claps: 10,
    liked: false,
    image: 'https://i.ibb.co/LNxt44v/abstract.png',
  },
];

jest.mock('../../../utils/makeRequest');
describe('CardContainer', () => {
  it('should render correctly', async () => {
    makeRequest.mockResolvedValue(mockData);
    const { asFragment } = render(<CardContainer />);
    await waitFor(() => {
      expect(screen.getByText('Mock Title 1')).toBeInTheDocument();
    });
    expect(asFragment).toMatchSnapshot();
  });
  it('should display error message when some error occurs', async () => {
    makeRequest.mockRejectedValue({ message: 'Error' });
    const { asFragment } = render(<CardContainer />);
    await waitFor(() => {
      expect(screen.getByText('Error')).toBeInTheDocument();
    });
    expect(asFragment).toMatchSnapshot();
  });
  it('should display loading message when data is being fetched', async () => {
    makeRequest.mockResolvedValue(mockData);
    render(<CardContainer />);
    expect(screen.getByText('Loading...')).toBeTruthy();
    await waitFor(() => {
      expect(screen.getByText('Mock Title 1')).toBeTruthy();
    });
  });
});
