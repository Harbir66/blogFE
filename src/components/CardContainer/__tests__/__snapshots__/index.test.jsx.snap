// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should display error message when some error occurs 1`] = `[Function]`;

exports[`CardContainer should render correctly 1`] = `[Function]`;
